server/sonar-server/src/main/java/org/sonar/server/es/StickyFacetBuilder.java

```diff
@@ -62,12 +63,21 @@ public StickyFacetBuilder(QueryBuilder query, Map<String, QueryBuilder> filters,
   }
 
   public AggregationBuilder buildStickyFacet(String fieldName, String facetName, Object... selected) {
-    return buildStickyFacet(fieldName, facetName, FACET_DEFAULT_SIZE, selected);
+    return buildStickyFacet(fieldName, facetName, t -> t, selected);
+  }
+
+  public AggregationBuilder buildStickyFacet(String fieldName, String facetName, Function<TermsBuilder, AggregationBuilder<?>> feature, Object... selected) {
+    return buildStickyFacet(fieldName, facetName, FACET_DEFAULT_SIZE, feature, selected);
   }
 
   public AggregationBuilder buildStickyFacet(String fieldName, String facetName, int size, Object... selected) {
+    return buildStickyFacet(fieldName, facetName, size, t -> t, selected);
+  }
+
+  private AggregationBuilder buildStickyFacet(String fieldName, String facetName, int size, Function<TermsBuilder, AggregationBuilder<?>> addAdditionalLayerIfNeeded,
```

Comment:
what is a "layer" ? "additionalSubAggregation" or "additionalAggregationFilter" ?